packages/@aws-cdk/cdk/lib/aspects/README.md

```diff
@@ -0,0 +1,292 @@
+## Aspects
+
+Aspects are a mechanism to extend the CDK without having to direclty impact the
+class heirarchy directly. This is a form of [Aspect-oriented
+programming](https://en.wikipedia.org/wiki/Aspect-oriented_programming) which we
+have implemented using the [Visitor
+Pattern](https://en.wikipedia.org/wiki/Aspect-oriented_programming).
+
+### Aspects in the CDK
+
+An aspect in the CDK is defined by the interface:
+
+```ts
+/**
+ * Represents an Aspect
+ */
+export interface IAspect {
+  /**
+   * Aspects invocation pattern
+   */
+  readonly visitType: AspectVisitType;
+  /**
+   * All aspects can visit a IConstruct
+   */
+  visit(node: IConstruct): void;
+}
+```
+
+Initially two types of aspects are supported. 
+
+1. Single visit aspects will ensure the aspect visit function is only invoked
+   one time.
+1. Multiple visit aspects will invoke the visit function as many times as
+   requested
+
+Aspects can be applied to any `Construct`. During the tree
+preparation phase the aspect will visit each construct in the tree at least one
+time. Aspects are invoked in the order they were added to the `Construct`,
+starting at the `App` (root of the tree) and progressing in order to the leaf
+nodes (most commonly the CloudFormation Resource). Aspect authors will implement 
+the `visit(IConstruct)` function can inspect the `Construct` for specific characteristics.
+Such as, is this construct a CloudFormation Resource?
+
+### Tag and Remove Tag Aspect
+
+Tag support is the first use of Aspects. The goal was to enable the ability to
```

Comment:
Not sure that people care whether or not Tags use Aspects, I think they only care about how tagging works.

Reply:
I think aspects are an important (albeit advanced concept) which @Doug-AWS  should definitely add to our developer guide.